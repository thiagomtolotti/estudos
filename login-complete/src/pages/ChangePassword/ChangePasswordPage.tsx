"use client";

import { useState } from "react";
import ChangePasswordForm from "./ChangePasswordForm";
import styles from "./ChangePasswordPage.module.css";
import ChangePasswordConfirm from "./ChangePasswordConfirm";
import { useChangePassword } from "@/hooks/useChangePassword";

export function ChangePasswordPage() {
	const [hasSetNewPassword, setHasSetNewPassword] = useState(false);

	const changePassword = useChangePassword();

	function sendNewPassword(newPassword: string) {
		changePassword(newPassword);

		setHasSetNewPassword(true);
	}

	return (
		<div className={styles.container}>
			<h1>Mudar a senha</h1>

			{!hasSetNewPassword ? (
				<ChangePasswordForm onSubmit={sendNewPassword} />
			) : (
				<ChangePasswordConfirm />
			)}
		</div>
	);
}
