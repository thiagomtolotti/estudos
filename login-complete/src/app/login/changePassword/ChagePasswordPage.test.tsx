import { render, RenderResult } from "@testing-library/react";
import ChangePasswordPage from "./page";

describe("Change Password page", () => {
	let page: RenderResult;

	beforeEach(() => {
		page = render(<ChangePasswordPage />);
	});

	it("Should render correctly", () => {
		expect(page.container).toMatchSnapshot();
	});
});
