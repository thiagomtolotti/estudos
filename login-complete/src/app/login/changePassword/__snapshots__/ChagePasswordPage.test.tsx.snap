// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change Password page Should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Mudar a senha
    </h1>
    <form>
      <input
        placeholder="Nova senha"
        type="password"
      />
      <input
        placeholder="Confirme a nova senha"
        type="password"
      />
      <span>
        <input
          data-testid="show-pass"
          id="show-pass"
          name="show-pass"
          type="checkbox"
        />
        <label
          for="show-pass"
        >
          Mostrar as senhas
        </label>
      </span>
      <button>
        Redefinir a senha
      </button>
    </form>
  </div>
</div>
`;
